src/function/gds/all_shortest_paths.cpp

```diff
@@ -133,50 +104,49 @@ class ParentPtrsAtomics {
         parentEdgePtr->setNextPtr(expected);
     }
 
-    std::atomic<ParentIterAndNextPtr*>* getCurFixedParentPtrs() {
-        return currFixedParentPtrs.load(std::memory_order_relaxed);
+    array_entry_t* getCurFixedParentPtrs() {
+        return currParentPtrs.load(std::memory_order_relaxed);
     }
 
-    void fixNodeTable(common::table_id_t tableID) {
-        KU_ASSERT(parentPtrs.contains(tableID));
-        currFixedParentPtrs.store(reinterpret_cast<std::atomic<ParentIterAndNextPtr*>*>(
-                                      parentPtrs.at(tableID).get()->buffer.data()),
-            std::memory_order_relaxed);
+    void pinNodeTable(common::table_id_t tableID) {
+        KU_ASSERT(objectArray.contains(tableID));
+        currParentPtrs.store(objectArray.getData(tableID), std::memory_order_relaxed);
     }
 
 private:
     std::mutex mtx;
     MemoryManager* mm;
-    common::table_id_map_t<std::unique_ptr<MemoryBuffer>> parentPtrs;
-    std::atomic<std::atomic<ParentIterAndNextPtr*>*> currFixedParentPtrs;
-    std::vector<std::unique_ptr<ParentPtrsBlock>> blocks;
+    ObjectArrayByTable<array_entry_t> objectArray;
```

Comment:
`objectArray` is too general. I'd call it `parents` or `bfsGraph`.

Reply:
Done